clients/src/main/java/org/apache/kafka/common/metrics/stats/Max.java

```diff
@@ -21,8 +21,15 @@
  */
 public final class Max extends SampledStat {
 
+    private final boolean whetherCanBeNegative;
+
     public Max() {
-        super(Double.NEGATIVE_INFINITY);
+        this(false);
```

Comment:
Having this as the default doesn't seem right. Not to mention backwards compatibility issues.